bundles/binding/org.openhab.binding.ddwrt/src/main/java/org/openhab/binding/ddwrt/internal/ddwrtBinding.java

```diff
@@ -0,0 +1,542 @@
+import org.quartz.CronScheduleBuilder;
+import org.quartz.CronTrigger;
+import org.quartz.Job;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.TriggerKey;
+import org.quartz.impl.StdSchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The ddwrt binding connects to a AVM ddwrt on the monitor port 1012 and
+ * listens to event notifications from this box. There are event for incoming
+ * and outgoing calls, as well as for connections and disconnections.
+ *
+ * @author Kai Kreuzer
+ * @author Jan N. Klug
+ * @author Markus Eckhardt
+ * @since 0.1.0
+ */
+public class ddwrtBinding extends AbstractActiveBinding<ddwrtBindingProvider> implements ManagedService {
+
+    private static HashMap<String, String> queryMap = new HashMap<>();
+
+    // TODO: configurable?
+    // daily cron schedule
+    private final String cronSchedule = "0 0 0 * * ?";
+
+    static {
+        queryMap.put(ddwrtBindingProvider.TYPE_ROUTER_TYPE, "nvram get DD_BOARD");
+        queryMap.put(ddwrtBindingProvider.TYPE_WLAN_24, "ifconfig");
+        queryMap.put(ddwrtBindingProvider.TYPE_WLAN_50, "ifconfig");
+        queryMap.put(ddwrtBindingProvider.TYPE_WLAN_GUEST, "ifconfig");
+    }
+
+    @Override
+    public void bindingChanged(BindingProvider provider, String itemName) {
+        super.bindingChanged(provider, itemName);
+
+        conditionalDeActivate();
+
+    }
+
+    private void conditionalDeActivate() {
+        logger.info("ddwrt conditional deActivate: {}", bindingsExist());
+
+        if (bindingsExist()) {
+            activate();
+        } else {
+            deactivate();
+        }
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(ddwrtBinding.class);
+
+    /* The IP address to connect to */
+    protected static String ip;
+
+    /* The port to connect to, default 23 */
+    protected static String port;
+
+    /* The password of the ddwrt to access via Telnet */
+    protected static String password;
+
+    /* The username, if used for telnet connections */
+    protected static String username;
+
+    /* The interface for 2.4 GHz network, e.g. ath0 */
+    protected static String interface_24;
+
+    /* The interface for 5.0 GHz network, e.g. ath1 */
+    protected static String interface_50;
+
+    /* The interface for guest network, e.g. virt_ath0 */
+    protected static String interface_guest;
+
+    /**
+     * Reference to this instance to be used with the reconnection job which is
+     * static.
+     */
+    private static ddwrtBinding INSTANCE;
+
+    public ddwrtBinding() {
+        INSTANCE = this;
+    }
+
+    @Override
+    public void activate() {
+        super.activate();
+        setProperlyConfigured(true);
+    }
+
+    @Override
+    public void deactivate() {
+    }
```

Comment:
Should something be done when this is deactivated, like stopping the Quartz jobs, disconnecting, etc.?